api/bindings.h

```diff
@@ -61,6 +61,22 @@ typedef struct cache_t {
 
 } cache_t;
 
+/**
+ * A view into an externally owned byte slice (Go `[]byte`).
+ * Use this for the current call only. A view cannot be copied for safety reasons.
+ * If you need a copy, use [`to_owned`].
+ *
+ * Go's nil value is fully supported, such that we can differentiate between nil and an empty slice.
+ */
+typedef struct ByteSliceView {
```

Comment:
Okay, so this is only for read-only slices coming from Go?
And everything we mutate or return needs to use Buffer still.

Makes sense to separate these cases - go data vs rust data

Reply:
> coming from Go

Basically yes, but I would stress "owned/managed by Go"